base/math.jl

```diff
@@ -37,7 +37,7 @@ end
     throw(DomainError(x,
         LazyString(f," was called with a real argument < -1 but will only return a complex result if called with a complex argument. Try ", f,"(Complex(x)).")))
 end
-@noinline function throw_exp_domainerror(x)
+@assume_effects :terminates_globally @noinline function throw_exp_domainerror(x)
```

Comment:
Is this really necessary? Throwing an error really should be known to terminate. Is LazyString getting in the way somehow? If so, IMO, this should be fixed on the LazyString side, not here.

Reply:
I had really to struggle with the compiler effects and I am not yet finished. This was an intermediate step and I will see, it it works without. (at the end we need ^(Float,Int) and ^(Float,Float) to be foldable.

Reply:
Seems not required - removed